packages/frontend/core/src/components/page-list/operation-cell.tsx

```diff
@@ -37,37 +44,63 @@ import type { AllPageListConfig } from './view';
 import { useEditCollection, useEditCollectionName } from './view';
 
 export interface PageOperationCellProps {
-  favorite: boolean;
-  isPublic: boolean;
-  link: string;
+  page: DocMeta;
   isInAllowList?: boolean;
-  onToggleFavoritePage: () => void;
-  onRemoveToTrash: () => void;
-  onDuplicate: () => void;
-  onDisablePublicSharing: () => void;
-  onOpenInSplitView: () => void;
   onRemoveFromAllowList?: () => void;
 }
 
 export const PageOperationCell = ({
-  favorite,
-  isPublic,
   isInAllowList,
-  link,
-  onToggleFavoritePage,
-  onRemoveToTrash,
-  onDuplicate,
-  onDisablePublicSharing,
-  onOpenInSplitView,
+  page,
   onRemoveFromAllowList,
 }: PageOperationCellProps) => {
   const t = useAFFiNEI18N();
+  const currentWorkspace = useService(Workspace);
   const { appSettings } = useAppSettingHelper();
+  const { setTrashModal } = useTrashModalHelper(currentWorkspace.docCollection);
   const [openDisableShared, setOpenDisableShared] = useState(false);
+  const favAdapter = useService(FavoriteItemsAdapter);
+  const favourite = useLiveData(
+    useMemo(() => favAdapter.isFavorite$(page.id, 'doc'), [favAdapter, page.id])
```

Comment:
should no need useMemo

Reply:
calling `favAdapter.isFavorite$` without memo will return a new observable on re-render, and then by combining with `useLiveData` there will be constant sub/unsubs under the hood. I think there will be some performance issues thus why I added use memo

Reply:
sub/unsub should be fast. This should not cause performance issues.